### ISSUE TITLE
SIGABRT when calling page.getTextWords()

### ISSUE THREAD

**Issue by asshatter** [rk700/PyMuPDF#165]
I'm getting SIGABRT Signal when I call page.getTextWords()
System and Environment Info : 
MacOS High Sierra, Python 2.7, RAM : 8 GB, PyMuPDF version : 1.12.6.

getText(), getTextBlocks() these all work but I get an abort signal when I call getTextWords for a specific page in a PDF. Can you check if there's no memory leak?

**Comment by JorjMcKie** [comment]
Sure. Can you please send me the PDF. If confidential, use my e-mail instead of this channel. Or create a 1-page PDF containing only the critical page (hoping that at least this page can be shared).
As a preliminary question: does that page contain words (= string with no intervening space) longer than 127 characters? If yes, I might try to simulate something ...

**Comment by JorjMcKie** [comment]
Hm ...
I have checked the function again and indeed found 2 issues. All words in the list are UTF-8 encoded (which is the case for all page.getText* methods).

1) the buffer for composing one word is a fixed storage area of length 128. If, for a very long word, a unicode character of length > 1 is unfortunate enough to start in the last byte of this area, an overflow will occur.
2) Words occupying more than 127 bytes in that buffer are split and output as separate words. Currently, the last byte of that are is not outputted in this siuation.

Given the confidentiality situation I have to assume you are somehow hit by this issues. I am testing a fix for this bug, will upload it (I believe today) and let you know.
Btw., I think now that 128 bytes are probably a little bit too churlish, and I am increasing this, too.

Thanks for reporting this!

**Comment by asshatter** [comment]
Thanks for looking it up. Please notify me once the issue is fixed.

**Comment by JorjMcKie** [comment]
Hi, fix is uploaded and merged into master.
I hope, the Travis wheel building jobs this time work for MacOS. If not you will have to generate the new version 1.13.1 manually for your Mac.

What I did concerning my 2 above points:
1) I am now creating the words in a dynamically allocated area, which automatically grows as required. So overflows should not occur again. Verified this with a 400 character unicode word.
2) As a consequence of the previous, words cannot get "too long" and will never be split into to shorter pieces.

As a drawback of this change, performance seems to suffer a little bit and the method appeared to need 2% more time than before (measured on my Windows machine, Python 3.6 x64). I think it's still very tolerable: processing all the 1310 pages of the Adobe manual happens in 2.5 seconds ...
